server/internal/interface/grpc/handlers/arkservice.go

```diff
@@ -145,11 +201,36 @@ func (h *handler) RegisterPayment(ctx context.Context, req *arkv1.RegisterPaymen
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
+	var cosigner *secp256k1.PublicKey
+
+	if h.svc.IsCovenantLess() {
```

Comment:
this is not really needed. We can just parse and the key if it's not nil, then at app level uses or ignores is depending on its version.